lib/irb/input-method.rb

```diff
@@ -181,10 +183,13 @@ def initialize
       @eof = false
 
       if Readline.respond_to?("basic_word_break_characters=")
-        Readline.basic_word_break_characters = IRB::InputCompletor::BASIC_WORD_BREAK_CHARACTERS
+        Readline.basic_word_break_characters = BASIC_WORD_BREAK_CHARACTERS
       end
       Readline.completion_append_character = nil
-      Readline.completion_proc = IRB::InputCompletor::CompletionProc
+      Readline.completion_proc = ->(target) {
+        bind = IRB.conf[:MAIN_CONTEXT].workspace.binding
+        RegexpCompletor.new.completion_candidates('', target, '', bind: bind)
```

Comment:
Why don't we store the completor in ivar for `ReadlineInput` as well?

Reply:
Nice catch! done